src/ESM.sol

```diff
@@ -26,25 +26,32 @@ interface GemLike {
 }
 
 interface EndLike {
+    function live() external view returns (uint256);
+    function vat()  external view returns (address);
     function cage() external;
 }
 
+interface VatLike {
+    function deny(address) external;
+}
+
 contract ESM {
-    GemLike public immutable gem; // collateral
-    EndLike public immutable end; // cage module
-    address public immutable pit; // burner
-    uint256 public immutable min; // threshold
-    uint256 public fired;
```

Comment:
I removed this variable as `end.cage()` can't be called twice, but if you consider it is better to have the redundant storage variable I'm ok about adding it back.

Reply:
It seems to protect excess MKR from being burned in a race to `join` sufficient stake and then `fire`. I'd agree there's not much point to this as either
a) MKR is going to zero anyway from an unrecoverable disaster
b) a new MKR token will be created to fork out the malicious actor(s), or if there was no malicious actor, fresh MKR can be `minted` (the pause proxy will still control the MkrAuthority).

Reply:
Yeah, anyway I'm adding `require(end.live() == 1, "ESM/system-already-shutdown");` to the `join` function, so it should act the same way `esm.fire()` did before.